### PR TITLE
refactor NewRobotAccount and ListRobotAccounts (robot client V2)

### PR DESCRIPTION
**Pull request by manhtukhang**
- `NewRobotAccount` method should return an additional `RobotCreated` in the result. Because the `Secret` of the created Robot Account could not retrieve later by Get methods, so, we need to know it after this method call.
  This resolve: #127 

- `ListRobotAccounts` method should return all robot accounts for the working correctly of `GetRobotAccountByName`, `DeleteRobotAccountByName` and `RefreshRobotAccountSecretByName` methods
  This resolve: #129

### CODE REVIEW

**Comment by manhtukhang** [comment]
Hi @elenz97,

Please help me review this PR.

Thanks!

File: apiv2/pkg/clients/robot/robot.go [hunk 1]
```diff
@@ -87,14 +87,31 @@ func (in AccessAction) String() string {
...
 // ListRobotAccounts ListProjectRobots returns a list of all robot accounts.
 func (c *RESTClient) ListRobotAccounts(ctx context.Context) ([]*modelv2.Robot, error) {
-	resp, err := c.V2Client.Robot.ListRobot(&robot.ListRobotParams{
-		Context: ctx,
-	}, c.AuthInfo)
-	if err != nil {
-		return nil, handleSwaggerRobotErrors(err)
+	var robotAccounts []*modelv2.Robot
+	var page int64 = c.Options.Page
+	var pageSize int64 = c.Options.PageSize
```
Comment (elenz97) [inline apiv2/pkg/clients/robot/robot.go:92]: When instantiating a new client e.g. via [`NewRESTClient()`](https://github.com/mittwald/goharbor-client/blob/master/apiv2/client.go#L88) you can pass an optional [`*config.Options`](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/config/options.go#L7). By this, the values for `page` or `pageSize` should be implicitly set when calling `ListRobotAccounts()`. See below comment for an example.

File: apiv2/pkg/clients/robot/robot.go [hunk 2]
```diff
@@ -87,14 +87,31 @@ func (in AccessAction) String() string {
...
+
+	for {
+		resp, err := c.V2Client.Robot.ListRobot(&robot.ListRobotParams{
+			Context:  ctx,
+			Page:     &page,
+			PageSize: &pageSize,
+		}, c.AuthInfo)
```
Comment (elenz97) [inline apiv2/pkg/clients/robot/robot.go:99]:
```suggestion
	resp, err := c.V2Client.Robot.ListRobot(&robot.ListRobotParams{
		Page:     &c.Options.Page,
		PageSize: &c.Options.PageSize,
		Q:        &c.Options.Query,
		Sort:     &c.Options.Sort,
		Context:  ctx,
	}, c.AuthInfo)
```

The wrapping `for`-loop could then be removed.

File: apiv2/pkg/clients/robot/robot.go (hunk 1 above)
Comment (manhtukhang) [reply apiv2/pkg/clients/robot/robot.go:92]: But in [GetRobotAccountByName()](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/clients/robot/robot.go#L103)/ [DeleteRobotAccountByName](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/clients/robot/robot.go#L146)/ [RefreshRobotAccountSecretByName](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/clients/robot/robot.go#L145) need to call `ListRobotAccounts()`. If *total robot accounts > pageSize* (which is enforced by the Goharbor API to maximum of 100) and *the matched robot account is >= .no 101*, `ListRobotAccounts()` couldn't list all, so above calling methods will return `errors.ErrRobotAccountUnknownResource`

Comment (elenz97) [reply apiv2/pkg/clients/robot/robot.go:92]: Thanks for pointing that out, @manhtukhang! I did not have in mind that the Goharbor API limits the maximum `PageSize` to 100 items (per request). Your solution might then also be applicable to other `List[...]` methods in this client. Anyhow, the call to `c.V2Client.Robot.ListRobot` should ideally contain the fields from [`*config.Options`](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/config/options.go#L7). If you could add those to your PR, i think it'll be ready to go! :+1:

**Review by elenz97 — CHANGES_REQUESTED** [review]

**Pull request merged**